test/public.spec.js

```diff
@@ -0,0 +1,423 @@
+    expect(window.NCC.imageChecker).toBeDefined();
+    expect(window.NCC.imageChecker.showImagesInfo).toEqual(jasmine.any(Function));
+    expect(window.NCC.imageChecker.hideImagesInfo).toEqual(jasmine.any(Function));
+    expect(window.NCC.imageChecker.isImagesInfoActive).toEqual(jasmine.any(Function));
+  });
+
+  describe('showImagesInfo()', function() {
+    it('should show image overlays only for available images', function() {
+      createDomNodes([
+        createMissingImg(),
+        createBackgroundImg(),
+        createNoBackgroundImg(),
+        createHiddenImg(),
+        createInvisibleImg(),
+        createNoSrcImg(),
+        createNoDimensionImg(),
+        createSmallImg(),
+        createMediumImg(),
+        createBigImg(),
+        createImg()
+      ]);
+
+      window.NCC.imageChecker.showImagesInfo();
+
+      let imageOverlays = document.querySelectorAll('.ncc-image-checker-overlay');
+      expect(imageOverlays.length).toEqual(5);
+    });
+
+    describe('big images overlays', function() {
+      beforeEach(function() {
+        createDomNodes([
+          createBigImg()
+        ]);
+        window.NCC.imageChecker.showImagesInfo();
+      });
+
+      it('should show all info', function() {
+        let textLines = getImageOverlayTextLines(0);
+
+        expect(textLines).toContain(jasmine.stringMatching('...holder-100x80.png'));
+        expect(textLines).toContain('Display: 200 x 160');
+        expect(textLines).toContain('Natural: 100 x 80');
+        expect(textLines).toContain('Image coverage: 25.00%');
+        expect(textLines).toContain('File Size: 4.464 KB');
+      });
+
+      it('should have url link to new tab', function() {
+        let imageOverlay = getImageOverlay(0);
+        let urlLink = imageOverlay.getElementsByTagName('a')[0];
+
+        expect(urlLink.href).toEqual(jasmine.stringMatching('test/assets/placeholder-100x80.png'));
+        expect(urlLink.target).toEqual('_blank');
+        expect(urlLink.innerText).toEqual(jasmine.stringMatching('...holder-100x80.png'));
+      });
+    });
+
+    describe('medium images overlays', function() {
+      beforeEach(function() {
+        createDomNodes([
+          createMediumImg()
+        ]);
+        window.NCC.imageChecker.showImagesInfo();
+      });
+
+      it('should not show url info', function() {
+        let imageOverlay = getImageOverlay(0);
+        let urlLink = imageOverlay.getElementsByTagName('a')[0];
+
+        expect(urlLink).toBeUndefined();
+      });
+
+      it('should show other info', function() {
+        let textLines = getImageOverlayTextLines(0);
+
+        expect(textLines).toContain('Display: 151 x 71');
+        expect(textLines).toContain('Natural: 100 x 80');
+        expect(textLines).toContain('Image coverage: 74.62%');
+        expect(textLines).toContain('File Size: 4.464 KB');
+      });
+
+      it('should click to new tab', function() {
+        let urlLink = getImageOverlay(0);
```

Comment:
Well, the thing is that for small images we add a link directly in the body, as the overlay while for the others we add a div with a link inside